ftw/slacker/slack_notifier.py

```diff
@@ -0,0 +1,87 @@
+from ftw.slacker.interfaces import ISlackNotifier
+from threading import Thread
+from zope.component import getUtility
+from zope.interface import implements
+import os
+import requests
+
+
+def notify_slack(*args, **kwargs):
+    """This is the main api mehtod to perform a slack notification
+
+    Always use this method to do slack notifications.
+
+    See the readme.rst to find out how to use it.
+    """
+    slacker = getUtility(ISlackNotifier)
+    return slacker.notify(*args, **kwargs)
+
+
+class SlackNotifier(object):
+    """The default slack notifier utility posts a message into slack through
+    a webhook.
+    """
+    implements(ISlackNotifier)
+
+    # Name of the request thread
+    THREAD_NAME = 'SlackNotifier-Thread'
+
+    # Use this string in your environment variables to deactivate
+    # notification.
+    NOTIFICATION_DEACTIVATION_VALUE = 'deactivate'
+
+    # Possible environment variables
+    STANDARD_SLACK_WEBHOOK = 'STANDARD_SLACK_WEBHOOK'
+    DEACTIVATE_SLACK_NOTIFICATION = 'DEACTIVATE_SLACK_NOTIFICATION'
+
+    def notify(self, webhook_url=None, timeout=2, verify=True, **payload):
+        """Performs the slack notification
+        """
+        if self._is_notifiaction_globally_deactivated():
+            return
+
+        webhook_url = self._choose_webhook_url(webhook_url)
+        if self._is_notification_deactivated(webhook_url):
+            return
+
+        thread = Thread(target=self._do_request,
+                        name=self.THREAD_NAME,
+                        args=(webhook_url, timeout, verify),
+                        kwargs=payload)
+
+        thread.start()
+        return thread
+
+    def _do_request(self, webhook_url=None, timeout=2, verify=True, **payload):
+        """Actually performs the request.
+        """
+        requests.post(webhook_url,
+                      timeout=timeout,
+                      verify=verify,
+                      json=payload,).raise_for_status()
```

Comment:
Jeas, it is. The main thread will not be blocked, but the console will raise the error and tell the user, that the error occured in a separate thread:

![bildschirmfoto 2017-07-26 um 15 01 43](https://user-images.githubusercontent.com/557005/28622240-21f5beec-7214-11e7-8fb7-351e15311ebe.png)

I have no tests for this because it's very hard to test and the value is very low.